lib/src/pubspec.dart

```diff
@@ -279,40 +283,48 @@ class Pubspec {
   /// The constraint on the Dart SDK, or [VersionConstraint.any] if none is
   /// specified.
   VersionConstraint get dartSdkConstraint {
-    _parseEnvironment();
+    _ensureEnvironment();
     return _dartSdkConstraint;
   }
 
   VersionConstraint _dartSdkConstraint;
 
   /// The constraint on the Flutter SDK, or `null` if none is specified.
   VersionConstraint get flutterSdkConstraint {
-    _parseEnvironment();
+    _ensureEnvironment();
     return _flutterSdkConstraint;
   }
 
   VersionConstraint _flutterSdkConstraint;
 
-  /// Parses the "environment" field and sets [_dartSdkConstraint] and
-  /// [_flutterSdkConstraint] accordingly.
-  void _parseEnvironment() {
+  /// Ensures that the top-level "environment" field has been parsed and
+  /// [_dartSdkConstraint] and [_flutterSdkConstraint] are set accordingly.
+  void _ensureEnvironment() {
     if (_dartSdkConstraint != null) return;
 
-    var yaml = fields['environment'];
+    var pair = _parseEnvironment(fields);
+    _dartSdkConstraint = pair.first;
+    _flutterSdkConstraint = pair.last;
+  }
+
+  /// Parses the "environment" field in [parent] and returns the Dart and
+  /// Flutter SDK constraints, respectively.
+  Pair<VersionConstraint, VersionConstraint> _parseEnvironment(YamlMap parent) {
```

Comment:
I think it's a little clearer that it's going to be exactly two constraints. I'd also feel a lot weirder making the positions in a list semantically meaningful than the positions in a pair.